### PR TITLE
Cleanup Hive views tests

### PR DESCRIPTION
**Pull request by losipiuk**
Extracted from https://github.com/trinodb/trino/pull/6720

### CODE REVIEW

File: testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestHiveViewsLegacy.java [hunk 1]
```diff
@@ -15,33 +15,27 @@
...
-import static java.lang.String.format;
 import static org.assertj.core.api.Assertions.assertThatThrownBy;
 
 @Requires({
         ImmutableNationTable.class,
         ImmutableOrdersTable.class,
 })
 public class TestHiveViewsLegacy
-        extends HiveProductTest
+        extends TestHiveViews
```
Comment (findepi) [inline testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestHiveViewsLegacy.java:38]: can we make an abstract class and avoid a test extending some other test?

Comment (losipiuk) [reply testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestHiveViewsLegacy.java:38]: Sure :)

**Comment by losipiuk** [comment]
@findepi good to go?

**Review by phd3 — APPROVED** [review]

File: testing/trino-product-tests/src/main/java/io/trino/tests/hive/AbstractTestHiveViews.java [hunk 2]
```diff
@@ -0,0 +1,353 @@
...
+                "VALUES " +
+                "  (CAST('two' AS varchar), ARRAY[11, 22]), " +
+                "  ('nothing', NULL), " +
+                "  ('zero', ARRAY[]), " +
+                "  ('one', ARRAY[42])");
+
+        onHive().executeQuery("DROP VIEW IF EXISTS hive_lateral_view");
+        onHive().executeQuery("CREATE VIEW hive_lateral_view as SELECT pageid, adid FROM pageAds LATERAL VIEW explode(adid_list) adTable AS adid");
+
+        assertViewQuery(
```
[the remark this answers is not in the document]
Comment (losipiuk) [reply testing/trino-product-tests/src/main/java/io/trino/tests/hive/AbstractTestHiveViews.java:147]: rabased. thx

File: testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestHiveViews.java [hunk 3]
```diff
@@ -13,245 +13,15 @@
...
-                queryAssert -> queryAssert.containsOnly(row(new BigDecimal("1.2345"), 42, "bar")));
-
-        assertThat(query("SELECT data_type FROM information_schema.columns WHERE table_name = 'hive_view_parametrized'")).containsOnly(
-                row("decimal(20,4)"),
-                row("bigint"),
-                row("varchar"));
-    }
-
     @Test(groups = HIVE_VIEWS)
     public void testSimpleCoral()
```
[the remark this answers is not in the document]
Comment (losipiuk) [reply testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestHiveViews.java:26]: makes sense

File: testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestHiveViewsLegacy.java [hunk 4]
```diff
@@ -130,18 +79,22 @@ public void testUnsupportedLateralViews()
                 .failsWithMessage("Failed parsing stored view 'hive.default.hive_lateral_view': line 1:78: mismatched input 'VIEW'");
     }
 
+    @Override
     @Test(groups = HIVE_VIEWS)
     public void testIdentifierThatStartWithDigit()
```
[the remark this answers is not in the document]
Comment (losipiuk) [reply testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestHiveViewsLegacy.java:84]: no - omission. Removing.

**Comment by losipiuk** [comment]
AC

File: testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestHiveViewsLegacy.java [hunk 5]
```diff
@@ -187,123 +126,26 @@ public void testHiveViewWithParametrizedTypes()
...
 
-    private static void assertViewQuery(String query, Consumer<QueryAssert> assertion)
+    @Override
+    public void testNestedHiveViews()
     {
-        // Ensure Hive and Presto view compatibility by comparing the results
-        assertion.accept(assertThat(onHive().executeQuery(query)));
-        assertion.accept(assertThat(onPresto().executeQuery(query)));
+        assertThatThrownBy(super::testNestedHiveViews)
+                .hasMessageContaining("iew 'hive.default.nested_top_view' is stale or in invalid state: column [n_renamed] of type varchar projected from query view at position 0 cannot be coerced to column [n_renamed] of type varchar(25) stored in view definition");
```
Comment (phd3) [inline testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestHiveViewsLegacy.java:136]: nit: iew --> View?

File: testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestHiveViewsLegacy.java [hunk 6]
```diff
@@ -15,40 +15,35 @@
...
 {
     @BeforeTestWithContext
     public void setup()
     {
         setSessionProperty("hive.legacy_hive_view_translation", "true");
     }
 
+    @Override
     @Test(groups = HIVE_VIEWS)
     public void testSelectOnView()
```
Comment (phd3) [inline testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestHiveViewsLegacy.java:48]: curious if https://github.com/trinodb/trino/pull/6721#discussion_r566478315 applies here

**Review by phd3 — APPROVED** [review]

File: testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestHiveViewsLegacy.java (hunk 5 above)
Comment (losipiuk) [reply testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestHiveViewsLegacy.java:136]: +1

File: testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestHiveViewsLegacy.java (hunk 6 above)
Comment (losipiuk) [reply testing/trino-product-tests/src/main/java/io/trino/tests/hive/TestHiveViewsLegacy.java]: Thanks. The test from super (despite reacher) works fine here too. Removed override.

**Pull request merged**